pkg/kube/util.go

```diff
@@ -177,6 +177,10 @@ func CheckPodReadyOrComplete(pod *corev1.Pod) error {
 func CheckPodReady(pod *corev1.Pod) error {
 	switch pod.Status.Phase {
 	case corev1.PodRunning:
+		// Delete endpoint when pod is going to be deleted.
```

Comment:
When a Pod is being deleted, it is shown as `Terminating` by some kubectl commands. This `Terminating` status is not one of the Pod phases.
Are you sure that it can be in `Running` state?

Reply:
kubectl will set DeletionTimestamp to non-null, so there is an intermediate state between them. Pod is still running normally when it is in `Running, Not Nil` state, and kubelet will send shutdown signal only when it is set to Terminating

Status,DeletionTimestamp
Running, nil
Running, Not Nil
Terminating, Not Nil

Reply:
I remember this logic is cipied from k8s

Reply:
Terminating is also a running phase，It a human friendly display.After Running state，pod phase must be Succeeded or Failed. It all means container is stop and no more restart.

k8s pod util only checks if ready condition true.But k8s endpoint ctrl will check is pod in deleting.
https://github.com/kubernetes/kubernetes/blob/8f2371bcceff7962ddb4901c36536c6ff659755b/pkg/controller/util/endpoint/controller_utils.go#L141

Reply:
This function is not used by pilot, have no effect on xds configuration

Reply:
Indeed, sorry about my rush behavior.I will check it again. 😵‍💫